Added like and dislike functionality #125

shobit1337 opened the pull request:
## Related Issue 

- Info about the related issue 

Closes: #125 

### Describe the changes you've made

I have added a like and dislike button functionality to the app and added a like counter, which will show the total likes of that post. Also added the styles for it in the style.js, everything will auto rendered on the change using the onSnapshot from firebase.

## Type of change

What sort of change have you made:
<!--
Example how to mark a checkbox:-
- [x] My code follows the code style of this project.
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mention any unusual behavior of your code (Write NA if not)
NA

## Checklist:
<!--
Example how to mark a checkbox:-
- [x] My code follows the code style of this project.
-->
- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.

## Screenshots
![image](https://user-images.githubusercontent.com/37979367/119195854-e8e93d00-baa2-11eb-9cbc-85637b93dc05.png)

## Additional Info (optional)
There are still lots of other bugs which is making the app unstable but I have tried to stick to my issue and successfully implemented the like/dislike functionality, I will create separate issues to fix other bugs.

abhishek2x commented on src/components/Post/index.js:
```diff
@@ -43,6 +81,24 @@ function Post({
+    .onSnapshot((snap) => {
+      let documents = [];
+      snap.forEach((doc) => {
+        documents.push({ userLiked: doc.id });
```
Since you are storing users who have liked a doc, you can also display them, by clicking the `no. of likes` button. I would suggest using a _modal_ to display them. Further, instead of `doc.id` store `doc.displayName`.

shobit1337 commented:
@abhishek2x I have implemented the changes you suggested, Also I have moved everything related to like into a new component LikePanel. Please review the changes.
![image](https://user-images.githubusercontent.com/37979367/119257047-34096a00-bbe1-11eb-87b2-0d2494504a51.png)

abhishek2x reviewed (approved):
LGTM

shobit1337 commented:
Thanks, abhishek2x,
@avinashkranjan can you take a look too
